nautobot/extras/tasks.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib.contenttypes.models import ContentType
+from django_rq import job
+
+from nautobot.extras.choices import CustomFieldTypeChoices
+
+
+@job("custom_fields")
+def update_custom_field_choice_data(field_id, old_value, new_value):
+    """
+    Update the values for a custom field choice used in objects' custom_field_data for the given field.
+
+    Args:
+        field_id (uuid4): The PK of the custom field to which this choice value relates
+        old_value (str): The existing value of the choice
+        new_value (str): The value which will be used as replacement
+    """
+    from nautobot.extras.models import CustomField
+
+    field = CustomField.objects.get(pk=field_id)
+
+    if field.type == CustomFieldTypeChoices.TYPE_SELECT:
+        # Loop through all field content types and search for values to update
+        for ct in field.content_types.all():
+            model = ct.model_class()
+            for obj in model.objects.filter(**{f"custom_field_data__{field.name}": old_value}):
+                obj.custom_field_data[field.name] = new_value
+                obj.save()
```

Comment:
What would be required to let us use `web_request_context` for change logging and webhooks on these tasks? Is it just a matter of making it so that `user` is an optional rather than mandatory parameter to that?

Reply:
Yep, that is pretty much it if we wanted to follow the present pattern. I have also wondered about a couple of packages out there that expose the request object wherever you want, akin to the way Flask works. I assume they work by throwing it in thread-local storage. (like this project which looks like someone's night time pet project https://github.com/schecterdamien/django-request-context).

Reply:
@jathanism @glennmatthews any further thoughts on this? We could do what Glenn suggested above but it would mean all those change log records never have a user associated.